rts/motoko-rts/src/gc/mark_compact/bitmap.rs

```diff
@@ -4,37 +4,47 @@ use crate::types::{size_of, Blob, Bytes, Obj};
 
 /// Current bitmap
 static mut BITMAP_PTR: *mut u8 = core::ptr::null_mut();
+static mut BITMAP_COMPENSATION: usize = 0;
+static mut BITMAP_SIZE: u32 = 0;
 
-pub unsafe fn alloc_bitmap<M: Memory>(mem: &mut M, heap_size: Bytes<u32>) {
+pub unsafe fn alloc_bitmap<M: Memory>(mem: &mut M, heap_size: Bytes<u32>, heap_prefix_words: u32) {
     // We will have at most this many objects in the heap, each requiring a bit
     let n_bits = heap_size.to_words().as_u32();
+    // Additional bits needed to account for
+    // unaligned heap start address
+    let prefix_bits = heap_prefix_words % 8;
```

Comment:
assert that `heap_prefix_words % 8 == 0`, i.e. the heap base is 32-byte aligned.

Reply:
then this can go away for good

Reply:
Gone in 0aaaeeb.